osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/test/java/org/opensmartgridplatform/adapter/protocol/iec60870/infra/messaging/DeviceRequestMessageListenerTest.java

```diff
@@ -41,8 +44,15 @@
     @Mock
     private DeviceResponseMessageSender deviceResponseMessageSender;
 
+    @BeforeEach
+    void setup() {
+        ReflectionTestUtils.setField(this.deviceRequestMessageListener, "correlationUidPerDevice",
+                new CorrelationUidPerDevice());
```

Comment:
It looks like this adds the CorreationUidPerDevice just to avoid exceptions when it is called.
Did you consider adding a mock instead and verifying in a new test that the correlation UID is enqueued for the proper device?
On the one hand this is covered by any (integration) test where the status is retrieved. On the other hand I wonder if unit testing this side, plus unit testing the class that dequeues correlation UIDs from CorrelationUidPerDevice would give some more certainty about the working of correlation UID matching when performing later changes to the code.